### PR TITLE
Documentation: building from source

### PR DESCRIPTION
**Pull request by apoorvedave1**
### What changes were proposed in this pull request?
Documentation: How to build hyperspace from source

### Why are the changes needed?
NA

### Does this PR introduce _any_ user-facing change?
NA

### How was this patch tested?
NA

### CODE REVIEW

File: docs/_docs/06-dg-building-from-source.md [hunk 1]
```diff
@@ -2,7 +2,39 @@
...
+To compile:
+```
+sbt compile
+```
+
+To run tests:
+```
+sbt test
+```
+### For Development On Intellij
```
Comment (imback82) [inline docs/_docs/06-dg-building-from-source.md:38]: nit: can we put same spacing (new lines) around ###?

File: docs/_docs/06-dg-building-from-source.md [hunk 2]
```diff
@@ -2,7 +2,39 @@
...
+1. Download and install [Intellij](https://www.jetbrains.com/idea/) with [Scala plugin](https://plugins.jetbrains.com/plugin/1347-scala) enabled
+2. Intellij -> Open -> Choose file `<root>/build.sbt` -> Open As Project
```
Comment (imback82) [inline docs/_docs/06-dg-building-from-source.md:40]: Can you add a step to compile and run tests using the sbt window? (to be consistent with CLI experience)

File: docs/_docs/06-dg-building-from-source.md [hunk 3]
```diff
@@ -2,7 +2,39 @@
 title: "Building from Source"
 permalink: /docs/dg-building-from-source/
 excerpt: "How to configure Hyperspace for your needs."
```
Comment (rapoth) [inline docs/_docs/06-dg-building-from-source.md:4]:
```suggestion
excerpt: "Building from source"
```

File: docs/_docs/06-dg-building-from-source.md [hunk 4]
```diff
@@ -2,7 +2,39 @@
...
+last_modified_at: 2020-06-23
 toc: true
 ---
 
+## Building from Source
+
+Hyperspace is built with [sbt](https://www.scala-sbt.org/). Run the following commands from the project root directory.
+
+### On *nix OS
+
```
Comment (imback82) [inline docs/_docs/06-dg-building-from-source.md:14]: Can you remove this section from the main README.md?

File: docs/_docs/06-dg-building-from-source.md (hunk 2 above)
Comment (apoorvedave1) [reply docs/_docs/06-dg-building-from-source.md:41]: added instructions to use sbt shell intellij for compile and test

File: docs/_docs/06-dg-building-from-source.md (hunk 1 above)
Comment (apoorvedave1) [reply docs/_docs/06-dg-building-from-source.md:39]: added. Thanks

File: docs/_docs/06-dg-building-from-source.md (hunk 4 above)
Comment (apoorvedave1) [reply docs/_docs/06-dg-building-from-source.md:14]: removed. Thanks @imback82

**Pull request merged**